### PR TITLE
Balance fix

### PR DESCRIPTION
**Pull request by liwenhui-soul**
#### What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

#### What problem(s) does this PR solve?
Issue(s) number: 

Description:

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by Sophie-Xie** [comment]
Which issues does this PR solve?

**Comment by HarrisChu** [comment]
fix https://github.com/vesoft-inc/nebula/issues/3654

File: src/meta/processors/job/BalanceJobExecutor.cpp [hunk 1]
```diff
@@ -88,6 +88,36 @@ nebula::cpp2::ErrorCode BalanceJobExecutor::save(const std::string& k, const std
   return rc;
 }
 
+void BalanceJobExecutor::insertOneTask(
+    const BalanceTask& task, std::map<PartitionID, std::vector<BalanceTask>>* existTasks) {
+  std::vector<BalanceTask>& taskVec = existTasks->operator[](task.getPartId());
```
Comment (kikimo) [inline src/meta/processors/job/BalanceJobExecutor.cpp:93]: Just curious, why not use reference instead?

Comment (liwenhui-soul) [reply src/meta/processors/job/BalanceJobExecutor.cpp:93]: > Just curious, why not use reference instead?

just follow google style, an output param should be non-const pointer

File: src/meta/processors/job/JobManager.cpp [hunk 2]
```diff
@@ -144,7 +144,7 @@ void JobManager::scheduleThread() {
 
 // @return: true if all task dispatched, else false
 bool JobManager::runJobInternal(const JobDescription& jobDesc, JbOp op) {
-  std::lock_guard<std::mutex> lk(muJobFinished_);
+  std::lock_guard<std::recursive_mutex> lk(muJobFinished_);
```
Comment (critical27) [inline src/meta/processors/job/JobManager.cpp:147]: Why we need to use recursive_mutex here?

**Review by critical27 — COMMENTED** [review]
Generally LGTM, maybe add some comments

File: src/meta/processors/job/JobManager.cpp (hunk 2 above)
Comment (liwenhui-soul) [reply src/meta/processors/job/JobManager.cpp:147]: if all balance tasks success and the job is stopped before job status updated, the balance tasks will get muJobFinished_ through call back in the same thread

Comment (liwenhui-soul) [reply src/meta/processors/job/JobManager.cpp:147]: have added comment at its declare place

**Comment by critical27** [comment]
Close #3672

**Review by critical27 — DISMISSED** [review]
Generally LGTM

File: src/kvstore/raftex/RaftexService.cpp [hunk 3]
```diff
@@ -155,10 +155,17 @@ void RaftexService::addPartition(std::shared_ptr<RaftPart> part) {
...
+  using FType = decltype(folly::makeFuture());
+  using FTValype = typename FType::value_type;
   // Stop the partition
-  part->stop();
+  folly::makeFuture()
+      .thenValue([this, &part](FTValype) {
+        folly::RWSpinLock::WriteHolder wh(partsLock_);
+        parts_.erase(std::make_pair(part->spaceId(), part->partitionId()));
+      })
+      .via(getThreadManager().get())
```
Comment (critical27) [inline src/kvstore/raftex/RaftexService.cpp:166]: Maybe add some comments here. Do you think we should use a global executor here?

Comment (liwenhui-soul) [reply src/kvstore/raftex/RaftexService.cpp]: done

**Review by liuyu85cn — DISMISSED** [review]

**Review by critical27 — APPROVED** [review]
Good job

**Pull request merged**